src/app/public/modules/wait/wait-adapter.service.ts

```diff
@@ -21,13 +36,95 @@ export class SkyWaitAdapterService {
     waitEl: ElementRef,
     isFullPage: boolean,
     isWaiting: boolean,
-    isNonBlocking = false
-  ): void {
+    id: string,
+    waitCmp: SkyWaitComponent
+  ) {
     let busyEl = isFullPage ? document.body : waitEl.nativeElement.parentElement;
     let state = isWaiting ? 'true' : undefined;
 
-    if (!isNonBlocking) {
+    if (!(waitCmp && waitCmp.isNonBlocking)) {
       this.renderer.setElementAttribute(busyEl, 'aria-busy', state);
+
+      // Remove focus from page when full page blocking wait
+      if (isFullPage || busyEl.contains(document.activeElement)) {
+        this.clearDocumentFocus();
+      }
+
+      if (isWaiting) {
+        // Prevent tab navigation within the waited page
+        let endListenerFunc = this.renderer.listen(document.body, 'keydown', (event: KeyboardEvent) => {
+          if (event.key.toLowerCase() === 'tab') {
+            event.preventDefault();
+            event.stopPropagation();
+            event.stopImmediatePropagation();
+            if (!waitCmp.isFullPage) {
+              // Propagate tab navigation if attempted into waited element
+              this.focusNextElement(busyEl, event.shiftKey);
+            } else {
+              this.clearDocumentFocus();
+            }
+          }
+        });
+        this.parentListeners[id] = endListenerFunc;
+      } else if (id in this.parentListeners) {
+        // Clean up existing listener
+        this.parentListeners[id]();
+        delete this.parentListeners[id];
+      }
+    }
+  }
+
+  private focusNextElement(parentElement: any, shiftKey: boolean): void {
+    // Select all possible focussable elements
+    let focussableElements =
+      'a[href], ' +
+      'area[href], ' +
+      'input:not([disabled]):not([tabindex=\'-1\']), ' +
+      'button:not([disabled]):not([tabindex=\'-1\']), ' +
+      'select:not([disabled]):not([tabindex=\'-1\']), ' +
+      'textarea:not([disabled]):not([tabindex=\'-1\']), ' +
+      'iframe, object, embed, ' +
+      '*[tabindex]:not([tabindex=\'-1\']), ' +
+      '*[contenteditable=true]';
+     let focussable = Array.prototype.filter.call(document.body.querySelectorAll(focussableElements),
+      (element: any) => {
+        return element.offsetWidth > 0 || element.offsetHeight > 0 || element === document.activeElement;
+      });
+     // If shift tab, go in the other direction
+    let modifier = shiftKey ? -1 : 1;
```

Comment:
Noticed something strange - try this:
1. Add a link to the element with text
2. Toggle the wait
3. Tab to the `Console Log Info` link in the header
4. Shift+Tab to go backwards
5. Notice the link inside the wait gets focus

Reply:
Ah yeah, I had a bad condition around this. Should be fixed now